### DISCUSSION TITLE
Get METAR from URL; Use raw data as integers for comparison with variables

### DISCUSSION THREAD

**Discussion by merius92** [python-metar/python-metar#138] [Q&A]
I’m having some issues getting the data from [here](https://tgftp.nws.noaa.gov/data/observations/metar/stations/). If I’m using something like the lines below it works just fine, but I haven’t found a way to get the METAR string via URL so far. Is there any function/method I should look into?

`
code = ("METAR KEWR 111851Z VRB03G19KT 2SM   R04R/3000VP6000FT TSRA BR FEW015 "
  "BKN040CB BKN065 OVC200 22/22   A2987 RMK AO2 PK WND 29028/1817 WSHFT "
  "1812 TSB05RAB22 SLP114 FRQ   LTGICCCCG TS OHD AND NW-N-E MOV NE P0013 "
  "T02270215"
  )
`
 
By using `station_file_url = http://www.noaa.gov/nsd_cccc.txt` I haven’t gotten any data, so my guess is that I’m missing something here.

On the other hand, is there an easy way to get `Visibility `data and `Cloud Ceiling Height` data as integer values in order to compare them against some other int variables? I feel like the approach I tried below might not be the most efficient.

`from metar import Metar
code = ("METAR KEWR 111851Z VRB03G19KT 2SM R04R/3000VP6000FT TSRA BR FEW015 "
    "BKN040CB BKN065 OVC200 22/22 A2987 RMK AO2 PK WND 29028/1817 WSHFT "
    "1812 TSB05RAB22 SLP114 FRQ LTGICCCCG TS OHD AND NW-N-E MOV NE P0013 "
    "T02270215")
obs = Metar.Metar(code)
vis = int(obs.visibility()[0])
vis > 1`

I'm sorry if my post is beyond the scope of this library and if the question is not properly asked (I just started coding last week as a hobby), I'm just curious if what I asked above is doable.

**Comment by akrherz** [comment]
> but I haven’t found a way to get the METAR string via URL so far

I am unsure what that statement means.  Typically python users use the excellent `requests` library to download resources by URL.

I think your code to get visibility is right, what makes you think it is wrong?

**Reply by merius92** [reply]
Thank you for the reply. I managed to pull the data with `requests`.
Regarding the visibility code, I was just wondering if there is an attribute in the library I can use to get the integer out of it. If there is not, I guess I'll just develop my approach in order to get the data I want.

On another note. Now I know this is way beyond the scope of this repo and conversation, but do you have any idea on this [website](https://tgftp.nws.noaa.gov/data/observations/metar/stations/) the METAR messages are being replaced by SPECI when there is the case? I tried checking this information by myself, but finding a SPECI message there is like finding the needle in the haystack.
I would like to write a code that works with both METAR and SPECI messages so I'd be interested to see how this library interacts with SPECI.

Thank you!

**Reply by NWS-NRC** [reply]
SPECI is a METAR, but normal METARs are within the HH:50 to HH:59 time frame so if you get new METARS outside of this time block they are most likely SPECI
